test/jasmine/tests/select_test.js

```diff
@@ -0,0 +1,183 @@
+var Plotly = require('@lib/index');
+var Lib = require('@src/lib');
+var DBLCLICKDELAY = require('@src/plots/cartesian/constants').DBLCLICKDELAY;
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var mouseEvent = require('../assets/mouse_event');
+
+
+describe('select box and lasso', function() {
+    var mock = require('@mocks/14.json');
+
+    afterEach(destroyGraphDiv);
+
+    function drag(path) {
+        var len = path.length;
+
+        mouseEvent('mousemove', path[0][0], path[0][1]);
+        mouseEvent('mousedown', path[0][0], path[0][1]);
+
+        path.slice(1, len).forEach(function(pt) {
+            mouseEvent('mousemove', pt[0], pt[1]);
+        });
+
+        mouseEvent('mouseup', path[len - 1][0], path[len - 1][1]);
+    }
+
+    function click(x, y) {
```

Comment:
See https://github.com/plotly/plotly.js/pull/176#discussion_r49474717

But yeah, I should mention why in the code too.